common/src/View/EntityAttributeGridTable.cpp

```diff
@@ -511,6 +511,57 @@ namespace TrenchBroom {
             m_showDefaultRows = showDefaultRows;
             update();
         }
+        
+        static StringSet allSortedAttributeNames(MapDocumentSPtr document) {
+            StringSet keySet;
+            const Model::AttributableNodeList nodes = document->allAttributableNodes();
+            for (const auto& node : nodes) {
+                for (const auto& attribute : node->attributes()) {
+                    keySet.insert(attribute.name());
+                }
+            }
+            return keySet;
+        }
+        
+        static StringSet allSortedValuesForAttributeNames(MapDocumentSPtr document, const StringList& names) {
```

Comment:
You should try to use the attribute name / value indices. You can query the indices for individual nodes and for the entire map (see World::m_attributableIndex). Furthermore, you can query numbered properties, and thereby enable autocompletion to consider target1, target2, etc.